api/src/test/java/gov/cms/ab2d/api/controller/AdminAPIIPTests.java

```diff
@@ -0,0 +1,224 @@
+package gov.cms.ab2d.api.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.okta.jwt.JwtVerificationException;
+import gov.cms.ab2d.api.SpringBootApp;
+import gov.cms.ab2d.common.dto.SponsorDTO;
+import gov.cms.ab2d.common.dto.SponsorIPDTO;
+import gov.cms.ab2d.common.model.Sponsor;
+import gov.cms.ab2d.common.repository.ContractRepository;
+import gov.cms.ab2d.common.repository.JobRepository;
+import gov.cms.ab2d.common.repository.RoleRepository;
+import gov.cms.ab2d.common.repository.SponsorRepository;
+import gov.cms.ab2d.common.repository.UserRepository;
+import gov.cms.ab2d.common.util.AB2DPostgresqlContainer;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.util.List;
+import java.util.Set;
+
+import static gov.cms.ab2d.common.util.Constants.ADMIN_PREFIX;
+import static gov.cms.ab2d.common.util.Constants.ADMIN_ROLE;
+import static gov.cms.ab2d.common.util.Constants.API_PREFIX;
+import static gov.cms.ab2d.common.util.Constants.SPONSOR_ROLE;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+@SpringBootTest(classes = SpringBootApp.class, webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@AutoConfigureMockMvc
+@Testcontainers
+public class AdminAPIIPTests {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Container
+    private static final PostgreSQLContainer postgreSQLContainer = new AB2DPostgresqlContainer();
+
+    @Autowired
+    private SponsorRepository sponsorRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private RoleRepository roleRepository;
+
+    @Autowired
+    private JobRepository jobRepository;
+
+    @Autowired
+    private ContractRepository contractRepository;
+
+    @Autowired
+    private TestUtil testUtil;
+
+    private String token;
+
+    private static final String IP_URL = "/ip";
+
+    @BeforeEach
+    public void setup() throws JwtVerificationException {
+        contractRepository.deleteAll();
```

Comment:
As per the slack discussion, mark this block with a todo, something like
`// todo Delete me once proper teardown is put in place.`